src/components/CodeHighlighter/CodeHighlighter.tsx

```diff
@@ -0,0 +1,14 @@
+import Prism from "prismjs";
+import "prismjs/themes/prism-okaidia.css";
+import "prismjs/components/prism-javascript.min.js";
+import "prismjs/components/prism-typescript.min.js";
+
+export const CodeBlock = ({ code, language }: { code: string; language: string }) => {
+	Prism.highlightAll();
```

Comment:
Put `Prism.highlightAll()` in a use-effect that runs when `code` changes.

```
useEffect(() => {
     Prism.highlightAll()
}, [code])
```

This way it wont re-run the highlighter on re-render; but will if code changes.